tests/cases/fourslash/getOccurrencesAbstract.ts

```diff
@@ -0,0 +1,20 @@
+/// <reference path='fourslash.ts' />
+
+////[|abstract|] class Animal {
+////    [|abstract|] walk(): void;
+////    [|abstract|] makeSound(): void;
+////}
+////// Abstract class below should not get highlighted
+////abstract class Foo {
+////    abstract foo(): void;
+////    abstract bar(): void;
+////}
+
+test.ranges().forEach(r => {

```

Comment:
Can you make this a for-of and cache the ranges once so that you don't have to call `test.ranges()` over every iteration?